components/collector/tests/source_collectors/dependency_track/test_security_warnings.py

```diff
@@ -101,6 +101,30 @@ async def test_filter_by_project_name_and_version(self):
         response = await self.collect(get_request_json_side_effect=[self.projects(), self.findings()])
         self.assert_measurement(response, value="1", entities=self.entities())
 
+    async def test_include_by_component_name(self):
+        """Test filtering by component name."""
+        self.set_source_parameter("components_to_include", ["other component"])
+        response = await self.collect(get_request_json_side_effect=[self.projects(), self.findings()])
+        self.assert_measurement(response, value="0", entities=[])
+
+    async def test_include_by_component_name_regular_expression(self):
+        """Test filtering by component name regular expression."""
+        self.set_source_parameter("components_to_include", ["component.*"])
+        response = await self.collect(get_request_json_side_effect=[self.projects(), self.findings()])
+        self.assert_measurement(response, value="1", entities=self.entities())
+
+    async def test_exclude_by_component_name(self):
+        """Test filtering by component name."""
+        self.set_source_parameter("components_to_ignore", ["component name"])
+        response = await self.collect(get_request_json_side_effect=[self.projects(), self.findings()])
+        self.assert_measurement(response, value="0", entities=[])
+
+    async def test_exclude_by_component_name_regular_expression(self):
+        """Test filtering by component name regular expression."""
+        self.set_source_parameter("components_to_ignore", ["other.*"])
+        response = await self.collect(get_request_json_side_effect=[self.projects(), self.findings()])
+        self.assert_measurement(response, value="1", entities=self.entities())
+
```

Comment:
The code allows for a combination of including and excluding, i.e. include `"python.*"` and ignore `".*2.7.18"` should probably select `["python-alpine", "python-2.7.18-devel"]`, but exclude `["alpine-python", "python-2.7.18"]`. Although this might not be true if we do partial string matching instead of full string

Reply:
We do a regular expression match (not a search), so it works as you describe.